src/cljs/harja/tiedot/urakka/laadunseuranta/talvihoitoreitit_tiedot.cljs

```diff
@@ -0,0 +1,230 @@
+                            #{}
+                            @valitut-kohteet-atom)
+          talvihoitoreitit (keep (fn [reitti] (when (contains? valitut-kohteet (:id reitti)) reitti)) @karttataso-talvihoitoreitit)
+          sijainnit (map (fn [rivi] (map :sijainti (:reitit rivi))) talvihoitoreitit)
+          extentit (harja.geo/extent-monelle (flatten sijainnit))
+          map-reitit (into [] (flatten (mapv (fn [reitti]
+                                               (mapv (fn [r]
+                                                       (when (:sijainti r)
+                                                         {:alue (merge {:tyyppi-kartalla :talvihoitoreitit
+                                                                        :stroke {:width 8
+                                                                                 :color (:vari reitti)}}
+                                                                  (:sijainti r))
+                                                          :extent (harja.geo/extent (:sijainti r))
+                                                          :tyyppi-kartalla :talvihoitoreitit
+                                                          :selite {:teksti "Talvihoitoreitti"
+                                                                   :img (pinni-ikoni "sininen")}
+                                                          :infopaneelin-tiedot {:pituus (:pituus r)
+                                                                                :nimi (:nimi reitti)}
+                                                          :ikonit [{:tyyppi :merkki
+                                                                    :paikka [:loppu]
+                                                                    :zindex 21
+                                                                    :img (pinni-ikoni "sininen")}]}))
+                                                 (:reitit reitti)))
+                                         talvihoitoreitit)))]
+      (when (and (not-empty talvihoitoreitit) (not (nil? talvihoitoreitit)) @karttataso-nakyvissa?)
+        (with-meta map-reitit
+          {:selitteet [{:vari (map :color asioiden-ulkoasu/talvihoitoreitit)
+                        :teksti "Talvihoitoreitit"
+                        :extent extentit}]
+           :extent extentit})))))
+
+;; Talvihoitoreittien excel-tiedoston lataamisessa käyttäjän informointiin tarkoitettu modal
+(defn tulos-modal [leikkaavat-virheet muut-virheet paivitetyt onnistuneet otsikko]
+  (let [otsikkotyyli (if (or (not (empty? onnistuneet)) (not (empty? paivitetyt))) :onnistunut :virhe)]
+    (modal/nayta!
+      {:otsikko otsikko
+       :otsikko-tyyli otsikkotyyli}
+      (when (or (not (empty? leikkaavat-virheet)) (not (empty? muut-virheet)) (not (empty? paivitetyt)) (not (empty? onnistuneet)))
+        [:div
+         (when (not (empty? leikkaavat-virheet))
```

Comment:
Näissä lopuissa not empty statementeissa taitaa toimia esim:
`(when (seq leikkaavat-virheet) ..`